jsonifier/models/models.py

```diff
@@ -0,0 +1,218 @@
+                            {"model": self._name, "func": str(function)},
+                        )
+                    continue
+            elif subparser:
+                if not (field.relational or field.type == "reference"):
+                    if strict:
+                        self._jsonify_bad_parser_error(field_name)
+                    if not tools.config["test_enable"]:
+                        _logger.error(
+                            "%(model)s.%(fname)s not relational",
+                            {"model": self._name, "fname": field_name},
+                        )
+                    continue
+                value = [
+                    self._jsonify_record(subparser, r, {}) for r in rec[field_name]
+                ]
+                if field.type in ("many2one", "reference"):
+                    value = value[0] if value else None
+            else:
+                resolver = field_dict.get("resolver")
+                value = rec._jsonify_value(field, rec[field.name])
+                value = resolver.resolve(field, rec)[0] if resolver else value
+
+            self._add_json_key(root, json_key, value)
+        return root
+
+    def jsonify(self, parser, one=False):
+        """Convert the record according to the given parser.
+
+        Example of (simple) parser:
+            parser = [
+                'name',
+                'number',
+                'create_date',
+                ('partner_id', ['id', 'display_name', 'ref'])
+                ('shipping_id', callable)
+                ('delivery_id', "record_method")
+                ('line_id', ['id', ('product_id', ['name']), 'price_unit'])
+            ]
+
+        In order to be consistent with the Odoo API the jsonify method always
+        returns a list of objects even if there is only one element in input.
+        You can change this behavior by passing `one=True` to get only one element.
+
+        By default the key into the JSON is the name of the field extracted
+        from the model. If you need to specify an alternate name to use as
+        key, you can define your mapping as follow into the parser definition:
+
+        parser = [
+             'field_name:json_key'
+        ]
+
+        """
+        if one:
+            self.ensure_one()
+        if isinstance(parser, list):
+            parser = convert_simple_to_full_parser(parser)
+        resolver = parser.get("resolver")
+
+        results = [{} for record in self]
+        parsers = {False: parser["fields"]} if "fields" in parser else parser["langs"]
+        records = self
+        if "jsonifier__date_user_tz" not in self.env.context:
+            # TODO: backward compat flag for v < 15.0
+            # Dates must always be UTC and the client/consumer of this data
+            # should deal w/ the format as preferred.
+            # Drop this flag for v15!
+            records = records.with_context(jsonifier__date_user_tz=True)
```

Comment:
@sebastienbeau  `jsonifier__date_user_tz` must be removed from the code and all the related code.